superset/common/query_context_factory.py

```diff
@@ -128,6 +128,8 @@ def _apply_granularity(
 
         if granularity := query_object.granularity:
             filter_to_remove = None
+            if is_adhoc_column(x_axis):  # type: ignore
```

Comment:
I'm curious why we need `# type: ignore` here?

Reply:
the x_axis is from form_data and it's ANY

Reply:
It may be a good idea to broaden the scope of the type guard to `Any` so it can handle arbitrary types.

Reply:
I think the type guard is better to scope down rather than scope up right?

Reply:
@zephyring @villebro I think this PR introduced another potential bug in Adhoc column as X-Axis, you should sent a probe query to ensure x-axis is a temporal column.

Reply:
hi @zhaoyongjie can you share more on what potential bug this may've introduced?